Dockerfile

```diff
@@ -1,92 +1,107 @@
-FROM ghcr.io/foundry-rs/foundry:master as builder
+# NOTE: The `latest` version of foundry does not include arm64 as a build platform, 
+#       but tagged versions do. Using an arm64-compatible image enables developers to 
+#       locally rebuild this image with different arguments for debugging/testing purposes.
+FROM ghcr.io/foundry-rs/foundry@sha256:4606590c8f3cef6a8cba4bdf30226cedcdbd9f1b891e2bde17b7cf66c363b2b3 AS node-builder
+# Copy only package.json and yarn.lock to nearly guarantee using a cached layer for node_modules.
+RUN apk add --no-cache npm && \
+  npm install -g yarn
+WORKDIR /app
+COPY ./package.json ./package.json
+COPY ./yarn.lock ./yarn.lock
+RUN yarn install --immutable 
 
+FROM ghcr.io/foundry-rs/foundry@sha256:4606590c8f3cef6a8cba4bdf30226cedcdbd9f1b891e2bde17b7cf66c363b2b3 as builder
+# Install node related depencies. The `npm` apk package includes node 18 as a dependency.
```

Comment:
Nit: I'd move this comment about the `FROM` and just use this to describe what the layer is doing.